### PR TITLE
Fixed some Clippy issues, Rustfmt and orbclient scroll/select fix

### PR DESCRIPTION
**Pull request by genusistimelord**
Fixed some Clippy issues, Rustfmt and made it so the editor-orbclient scrolls and selects like a real textbox.

### CODE REVIEW

**Comment by genusistimelord** [comment]
closing due to rustfmt.

File: examples/editor-orbclient/src/main.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ fn main() {
         1024 * display_scale as u32,
         768 * display_scale as u32,
         &format!("COSMIC TEXT - {}", font_system.locale),
-        &[WindowFlag::Resizable],
+        &[WindowFlag::Resizable, WindowFlag::Async],
```
Comment (jackpot51) [inline examples/editor-orbclient/src/main.rs:29]: Why are you enabling async? That will spin even when there are no events.

**Comment by jackpot51** [comment]
I'm going to revert some of the formatting changes and squash that into your commit.

**Comment by jackpot51** [comment]
Replaced by #20

**Pull request closed without merging**